test/machineslib.py

```diff
@@ -226,6 +226,46 @@ def get_next_mac(self, last_mac):
         new_mac = ':'.join(parts)
         return new_mac
 
+    def setup_mock_server(self, mock_server_filename, subj_names):
+        self.machine.upload(["files/min-openssl-config.cnf", f"files/{mock_server_filename}"], self.vm_tmpdir)
+
+        self.restore_file("/etc/hosts")
+        for i, sub_name in enumerate(subj_names):
+            self.machine.execute(f'echo "127.0.0.1 {sub_name}" >> /etc/hosts')
+            self.machine.execute(f'echo "DNS.{i + 2} = {sub_name}" >> {self.vm_tmpdir}/min-openssl-config.cnf')
```

Comment:
`self.write_file(..., append=True)` avoids the `>>` and restores automatically. For the second one you can use `self.machine.write(path, content, append=True)`.